src/Microsoft.IdentityModel.Tokens/DeflateCompressionProvider.cs

```diff
@@ -0,0 +1,105 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.IdentityModel.Logging;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// A compression provider that supports compression and decompression using the <see cref="CompressionAlgorithms.Deflate"/> algorithm.
+    /// </summary>
+    public class DeflateCompressionProvider : CompressionProvider
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeflateCompressionProvider"/> class used to compress and decompress used the <see cref="CompressionAlgorithms.Deflate"/> algorithm.
+        /// </summary>
+        public DeflateCompressionProvider()
```

Comment:
I would encourage you to review the [DeflateStream](https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream) documentation to fully understand the default options it is using. I think it could be useful to provide a property or constructor input parameter that enables the consumer of this library to specify the desired CompressionLevel. This would enable them to decide if they care more about speed or optimization of the compression being performed.

Reply:
This is a good idea! 👍 It looks like by default CompressionLevel is set to Optimal when the CompressionMode is Compress. I added in another constructor that allows a user to specify the desired compression level if they would like to.